codegen/proto/options.go

```diff
@@ -236,23 +238,34 @@ func ParseOptions(fileDescriptors []*collector.DescriptorWithPath) (Options, err
 }
 
 func getFieldOptions(field *descriptor.FieldDescriptorProto) (FieldOptions, error) {
-	validationDisabled, err := getFieldOptionOpenAPIValidationDisabled(field)
+	validationDisabled, jsonFieldNameOverride, err := getFieldOptionValues(field)
 	if err != nil {
 		return FieldOptions{}, err
 	}
 	return FieldOptions{
 		Field:                     field,
 		OpenAPIValidationDisabled: validationDisabled,
+		JsonFieldNameOverride:     jsonFieldNameOverride,
 	}, nil
 }
-func getFieldOptionOpenAPIValidationDisabled(field *descriptor.FieldDescriptorProto) (bool, error) {
+
+func getFieldOptionValues(field *descriptor.FieldDescriptorProto) (bool, string, error) {
 	cueOptRaw, err := proto.GetExtension(field.Options, cue.E_Opt)
 	if err == nil {
 		cueOpt, ok := cueOptRaw.(*cue.FieldOptions)
 		if !ok {
-			return false, eris.Errorf("internal error: invalid option type %T expecting *cueproto.FieldOptions", cueOpt)
+			return false, "", eris.Errorf("internal error: invalid option type %T expecting *cueproto.FieldOptions", cueOpt)
 		}
-		return cueOpt.DisableOpenapiValidation, nil
+		return cueOpt.DisableOpenapiValidation, cueOpt.JsonFieldNameOverride, nil
```

Comment:
nit: instead of returning each option as a separate arg, what about a struct with all opts?